src/exoticatechnologies/modifications/exotics/impl/GuardianShield.kt

```diff
@@ -0,0 +1,743 @@
+                                    val momentum = amount * -0.5f * 1 / momentumFactor
+                                    ForceApplier.applyMomentum(it.getParentStation(), collision, Vector2f.sub(ship.location, it.getLocation(), null), momentum, true)
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            fun transferFlux(amount: Float, workingMode: FluxTransferMode = FluxTransferMode.ONE_WAY, transfer: Boolean = false) {
+                /**
+                 * I don't know where i'm going with this two-way heat exchange
+                 * the whole concept of working with percentages is kinda broken at it's core:
+                 *  - the shield drone has 175k flux capacity
+                 *  - the host ship has whatever, lets say 25k or 8k.
+                 *  - 10% of shield drone's capacity (17.5k) will transform to 2.5k or 800 flux. 15k flux will have been lost.
+                 *
+                 *  on the other hand, if the ship has amassed 90% flux:
+                 *  - the 90% of let's say 25k is 22.5 will transform to 153k flux. 130.5k flux will have been magically created
+                 *
+                 *  I think the two-way exchange does kinda work, but exchange is the wrong word for it.
+                 */
+                drone?.let {
+                    log("--> transferFlux(amount=$amount, workingMode=$workingMode, transfer=$transfer)", "$LOGTAG:FluxTransfer")
+                    // should be between 0-100%
+                    val droneFluxTracker = it.fluxTracker
+                    val droneSoftFluxPercentageLevel = (droneFluxTracker.currFlux - droneFluxTracker.hardFlux) / droneFluxTracker.maxFlux
+                    val droneHardFluxPercentageLevel = droneFluxTracker.hardFlux / droneFluxTracker.maxFlux
+
+                    val shipFluxTracker = ship.fluxTracker
+                    val shipSoftFluxPercentageLevel = (shipFluxTracker.currFlux - shipFluxTracker.hardFlux) / shipFluxTracker.maxFlux
+                    val shipHardFluxPercentageLevel = shipFluxTracker.hardFlux / shipFluxTracker.maxFlux
+                    // Check the host-ship's flux levels.
+                    // If they're different from the shield-host's levels, apply the sharing code to both of them.
+                    log("droneSoftFluxPercentageLevel: ${droneSoftFluxPercentageLevel}, droneHardFluxPercentageLevel: ${droneHardFluxPercentageLevel}, drone hard flux: ${droneFluxTracker.hardFlux}, drone current flux: ${droneFluxTracker.currFlux}, drone max flux: ${droneFluxTracker.maxFlux}", "$LOGTAG:FluxTransfer")
+                    log("shipSoftFluxPercentageLevel: ${shipSoftFluxPercentageLevel}, shipHardFluxPercentageLevel: ${shipHardFluxPercentageLevel}, ship hard flux: ${shipFluxTracker.hardFlux}, ship current flux: ${shipFluxTracker.currFlux}, ship max flux: ${shipFluxTracker.maxFlux}", "$LOGTAG:FluxTransfer")
```

Comment:
comment these two out. some of these logs are quite spammy, and no longer necessary - but are still useful for keeping around for 'just in case'.

Reply:
done